### PR TITLE
Added support for installing puppet collections

### PR DESCRIPTION
**Pull request by grafjo**

### CODE REVIEW

**Comment by annawake** [comment]
Please do not add a special version type just for puppet collections - stick to the CM & CM_VERSION already there.  In this case, the CM type would possibly be something like PUPPET_COLLECTIONS and the CM_VERSION can be PC1, PC2, ... latest...or whatever...

**Comment by grafjo** [comment]
hi @annawake,
I'm aware of that. The thing is, with just CM & CM_VERSION you can not set the version for puppet-agent, e.g. building an image again will possible result in installing a different version of puppet-agent.

**Comment by annawake** [comment]
Why is this an issue?  From what I see, if you want to fix the version of the puppet agent, you would use CM=puppet, CM_VERSION=x.y.z and not bother with puppet collections (and use the puppet agent to install the puppet collection of your choosing)

**Comment by grafjo** [comment]
it's related to the new packaging structure of puppet - see https://puppetlabs.com/blog/welcome-puppet-collections
so when it's important to build predicable images, you can not do this without a given version of puppet collections _and_ a version of the puppet agent.

**Comment by annawake** [comment]
Sort out a way to do this by using the existing CM and CM_VERSION variables, please

**Comment by grafjo** [comment]
so `CM_VERSION=pc1;1.2.2` and `$(subst...)` is a better solution?

**Comment by grafjo** [comment]
@misheska what do you think?

File: script/cmtool.sh [hunk 1]
```diff
@@ -78,6 +79,23 @@ install_puppet()
     fi
 }
 
+install_puppet_collections()
+{
+    echo "==> Installing Puppet Collections ${CM_PC_VERSION}"
+    REDHAT_MAJOR_VERSION=$(egrep -Eo 'release ([0-9][0-9.]*)' /etc/redhat-release | cut -f2 -d' ' | cut -f1 -d.)
+
+    echo "==> Installing Puppet Labs repositories"
+    rpm -ipv "http://yum.puppetlabs.com/puppetlabs-release-${CM_PC_VERSION}-el-${REDHAT_MAJOR_VERSION}.noarch.rpm"
```
Comment (ruslansagitov) [inline script/cmtool.sh:88]: Use `https` here, as used in the official guide, [puppetlabs_package_repositories.html](https://docs.puppetlabs.com/guides/puppetlabs_package_repositories.html).

**Comment by annawake** [comment]
We're going to phase out the cmtool scripts - we never build them officially or publish them to atlas, so they are not being properly vetted anyway. Instead we'll be using custom-script.sh as the hook for actions like this, similar to what is being done in the ubuntu boxes.

**Pull request closed without merging**